src/plugins/midi-piano/midi-piano-editor.js

```diff
@@ -0,0 +1,197 @@
+import React, { Fragment, useState } from 'react';
+import { Form, Input, Radio, Button } from 'antd';
+import { sectionEditorProps } from '../../ui/default-prop-types.js';
+import { CDN_URL_PREFIX, MIDI_SOURCE_TYPE } from '../../domain/constants.js';
+import ResourcePicker from '../../components/resource-picker/resource-picker.js';
+import { storageLocationPathToUrl, urlToStorageLocationPath } from '../../utils/storage-utils.js';
+
+export default function MidiPianoEditor({ content, onContentChanged }) {
+
+  const FormItem = Form.Item;
+  const RadioGroup = Radio.Group;
+  const RadioButton = Radio.Button;
+  const { t } = useTranslation('midiPiano');
+  const { sourceType, sourceUrl, midiTrackTitle, activeNotes } = content;
+  const [keyRangeSelectorPianoOpened, setKeyRangeSelectorOpened] = useState(false);
+
+  const formItemLayout = {
+    labelCol: { span: 4 },
+    wrapperCol: { span: 14 }
+  };
+
+  const changeContent = newContentValues => {
+    const newContent = { ...content, ...newContentValues };
+    const isInvalidSourceUrl
+      = newContent.sourceType === MIDI_SOURCE_TYPE.external
+      && validation.validateUrl(newContent.sourceUrl, t).validateStatus === 'error';
+    onContentChanged(newContent, isInvalidSourceUrl);
+  };
+
+  const handleSourceTypeValueChanged = event => {
+    const { value } = event.target;
+    changeContent({ sourceType: value, sourceUrl: '' });
+  };
+
+  const handleExternalSourceUrlValueChanged = event => {
+    const { value } = event.target;
+    changeContent({ sourceUrl: value });
+  };
+
+  const handleInternalSourceUrlValueChanged = event => {
+    const { value } = event.target;
+    changeContent({ sourceUrl: value });
+  };
+
+  const handleInternalSourceUrlFileNameChanged = value => {
+    changeContent({ sourceUrl: value });
+  };
+
+  const handleMidiTrackTitleValueChanged = event => {
+    const { value } = event.target;
+    changeContent({ midiTrackTitle: value });
+  };
+
+  const handleActiveNotesChanged = midiValue => {
+    let validNoteRange;
+    let array = [...activeNotes];
+
+    if (array.includes(midiValue)) {
+      const index = array.indexOf(midiValue);
+      array.splice(index, 1);
+      if (array.length === 2) {
+        array.sort((a, b) => {
+          return a - b;
+        });
+        validNoteRange = {
+          first: array[0],
+          last: array[1]
+        };
+        changeContent({ activeNotes: array, noteRange: validNoteRange });
+        return;
+      }
+      changeContent({ activeNotes: array });
+      return;
+    }
+    array = [...activeNotes, midiValue];
+    if (array.length === 2) {
+      array.sort((a, b) => {
+        return a - b;
+      });
+      validNoteRange = {
+        first: array[0],
+        last: array[1]
+      };
+      changeContent({ activeNotes: array, noteRange: validNoteRange });
+      return;
+    }
+    changeContent({ activeNotes: array, noteRange: validNoteRange });
+  };
```

Comment:
I got rid of the key range selector for now, since there were problems with the mouse event handlers of the react-piano. I will remake it once we've decided on how to go on with the plugin. That is why I also got rid of this method for now.